src/components/views/rooms/LinkPreviewWidget.tsx

```diff
@@ -68,6 +69,17 @@ export default class LinkPreviewWidget extends React.Component<IProps> {
             link: this.props.link,
         };
 
+        if (this.image.current) {
+            const clientRect = this.image.current.getBoundingClientRect();
```

Comment:
We should have image's width and height available already, is it necessary to calculate it from the DOM instead? I don't necessarily understand why it's done this way here, so I may just be in the wrong. 😄

Reply:
I don't think we have the DOM element size anywhere (or do we? what am I missing?). We need to use this instead of using the image size as this is the real thumbnail size used for the animation

Reply:
We should have those available via const content = this.props.mxEvent.getContent<IMediaEventContent>(), see src/components/views/messages/MImageBody.tsx:321. If that won't work just resolve this comment please. 😄

Reply:
We need to know the real size of the element, not just the thumbnail image size as the real thumbnail element can be smaller than the thumbnail image.